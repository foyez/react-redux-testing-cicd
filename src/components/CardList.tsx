import React from 'react'

import { Card } from './Card'
import { Robot } from '../interfaces'

interface ICardList {
  robots: Robot[]
}

export const CardList: React.FC<ICardList> = ({ robots }) => (
  <div>
    {robots.map((robot) => {
      return <Card key={robot.id} {...robot} />
    })}
  </div>
)
