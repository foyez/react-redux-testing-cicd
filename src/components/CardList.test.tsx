import React from 'react'
import { shallow } from 'enzyme'

import { CardList } from './CardList'

describe('Card Component', () => {
  const robots = [{ name: 'foyez', email: 'foyez@email.com', id: 1 }]

  it('expects to render Card component', () => {
    const wrapper = shallow(<CardList robots={robots} />)

    expect(wrapper).toMatchSnapshot()
  })
})
