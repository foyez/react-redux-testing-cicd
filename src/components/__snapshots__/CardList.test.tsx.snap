// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component expects to render Card component 1`] = `ShallowWrapper {}`;
